msi-ec.c

```diff
@@ -2381,6 +2381,83 @@ static struct msi_ec_conf CONF28 __initdata = {
 	},
 };
 
+static const char *ALLOWED_FW_29[] __initconst = {
+	"16Q4EMS1.110", // GS65 Stealth
+	NULL
+};
+
+static struct msi_ec_conf CONF29 __initdata = {
+	.allowed_fw = ALLOWED_FW_29,
+	.charge_control = {
+		.address      = 0xef,
+		.offset_start = 0x8a,
+		.offset_end   = 0x80,
+		.range_min    = 0x8a,
+		.range_max    = 0xe4,
+	},
+	.webcam = {
+		.address       = 0x2e,
+		.block_address = MSI_EC_ADDR_UNSUPP,
+		.bit           = 1,
+	},
+	.fn_win_swap = {
+		.address = 0xbf,
+		.bit     = 4, // 0x00-0x10
+	},
+	.cooler_boost = {
+		.address = 0x98,
+		.bit     = 7,
+	},
+	.shift_mode = {
+		.address = 0xf2,
+		.modes = {
+			{ SM_ECO_NAME,     0xc2 }, // super battery
+			{ SM_COMFORT_NAME, 0xc1 }, // balanced
+			{ SM_TURBO_NAME,   0xc4 }, // extreme
+			{ SM_SPORT_NAME,   0xc0 }, // sport
+			MSI_EC_MODE_NULL
+		},
+	},
+	.super_battery = {
+		.address = MSI_EC_ADDR_UNSUPP, // Function not shown in dragon center
+	},
+	.fan_mode = {
+		.address = 0x89,
```

Comment:
This is unusual, can you check 0xf4 and 0xd4?

Reply:
0x89 is GPU realtime %RPM so it's you really wrong config

Reply:
It is surprisingly the address of the register that changes when I change the fan mode in windows dragon center. I already checked twice as it was suspicious but I will check again

Reply:
You need to look on f4

Reply:
It was f4 indeed